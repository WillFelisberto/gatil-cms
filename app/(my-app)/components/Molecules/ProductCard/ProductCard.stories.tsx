import type { Meta, StoryObj } from '@storybook/react';

import { ProductCard } from './ProductCard';

const meta: Meta<typeof ProductCard> = {
  title: 'Molecules/ProductCard',
  component: ProductCard,
  tags: ['autodocs']
};

export default meta;
type Story = StoryObj<typeof ProductCard>;

export const Default: Story = {
  args: {
    name: 'Caneca do Gatil',
    description: 'Ajude um resgatado com estilo! ☕',
    price: 39.9,
    image: {
      createdAt: '2023-01-01',
      updatedAt: '2023-01-01',
      id: '1',
      url: '/gato.jpg',
      alt: 'Caneca solidária'
    },
    buyLink: 'https://www.mercadopago.com.br/caneca'
  }
};
