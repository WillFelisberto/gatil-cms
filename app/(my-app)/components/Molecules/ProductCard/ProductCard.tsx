import Image from 'next/image';
import Link from 'next/link';

import type { Media } from '@/payload-types';

type ProductCardProps = {
  name: string;
  description?: string;
  price: number;
  image: Media;
  buyLink: string;
};

export const ProductCard = ({ name, description, price, image, buyLink }: ProductCardProps) => {
  const formattedPrice = new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL'
  }).format(price);

  return (
    <div
      className="bg-white rounded-2xl shadow-lg overflow-hidden flex flex-col"
      data-testid="product-card"
    >
      <div className="relative w-full h-64">
        <Image
          src={image?.url || ''}
          alt={image?.alt || name}
          fill
          className="object-cover"
          sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
        />
      </div>
      <div className="p-6 flex-1 flex flex-col justify-between gap-4">
        <div>
          <h3 className="text-xl font-semibold text-[#013274] mb-1" data-testid="product-name">
            {name}
          </h3>
          {description && (
            <p className="text-sm text-gray-700" data-testid="product-description">
              {description}
            </p>
          )}
        </div>
        <div className="flex flex-col gap-2">
          <span className="text-lg font-bold text-pink-600" data-testid="product-price">
            {formattedPrice}
          </span>
          <Link
            href={buyLink}
            target="_blank"
            rel="noopener noreferrer"
            className="inline-block text-center bg-pink-600 hover:bg-pink-700 text-white font-medium py-2 px-4 rounded-xl transition-all"
            data-testid="product-cta"
          >
            Comprar
          </Link>
        </div>
      </div>
    </div>
  );
};
