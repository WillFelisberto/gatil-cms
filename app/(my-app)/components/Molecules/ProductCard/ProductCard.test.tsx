import { render, screen } from '@/tests/test-utils';

import { ProductCard } from './ProductCard';

describe('<ProductCard />', () => {
  const mockProps = {
    name: 'Chaveiro Gatil',
    description: 'Produto solid치rio para apoiar o Gatil!',
    price: 25.5,
    buyLink: 'https://www.mercadopago.com.br/chaveiro',
    image: {
      createdAt: '2023-01-01',
      updatedAt: '2023-01-01',
      id: 1,
      url: '/chaveiro.jpg',
      alt: 'Chaveiro solid치rio'
    }
  };

  it('Should renders product name, description, price and button', () => {
    const { container } = render(<ProductCard {...mockProps} />);

    expect(screen.getByTestId('product-name')).toHaveTextContent('Chaveiro Gatil');
    expect(screen.getByTestId('product-description')).toHaveTextContent('Produto solid치rio');
    expect(screen.getByTestId('product-price')).toHaveTextContent(/25,50/i); // conforme Intl.format
    expect(screen.getByTestId('product-cta')).toHaveAttribute('href', mockProps.buyLink);

    expect(container).toMatchSnapshot();
  });

  it('Should renders image with correct alt text', () => {
    render(<ProductCard {...mockProps} />);
    const img = screen.getByAltText('Chaveiro solid치rio');
    expect(img).toBeInTheDocument();
  });
});
