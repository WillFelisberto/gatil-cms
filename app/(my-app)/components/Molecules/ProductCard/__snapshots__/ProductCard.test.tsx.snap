// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> Should renders product name, description, price and button 1`] = `
<div>
  <div
    class="bg-white rounded-2xl shadow-lg overflow-hidden flex flex-col"
    data-testid="product-card"
  >
    <div
      class="relative w-full h-64"
    >
      <img
        alt="Chaveiro solidário"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
        src="/_next/image?url=%2Fchaveiro.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fchaveiro.jpg&w=256&q=75 256w, /_next/image?url=%2Fchaveiro.jpg&w=384&q=75 384w, /_next/image?url=%2Fchaveiro.jpg&w=640&q=75 640w, /_next/image?url=%2Fchaveiro.jpg&w=750&q=75 750w, /_next/image?url=%2Fchaveiro.jpg&w=828&q=75 828w, /_next/image?url=%2Fchaveiro.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fchaveiro.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fchaveiro.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fchaveiro.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fchaveiro.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="p-6 flex-1 flex flex-col justify-between gap-4"
    >
      <div>
        <h3
          class="text-xl font-semibold text-[#013274] mb-1"
          data-testid="product-name"
        >
          Chaveiro Gatil
        </h3>
        <p
          class="text-sm text-gray-700"
          data-testid="product-description"
        >
          Produto solidário para apoiar o Gatil!
        </p>
      </div>
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="text-lg font-bold text-pink-600"
          data-testid="product-price"
        >
          R$ 25,50
        </span>
        <a
          class="inline-block text-center bg-pink-600 hover:bg-pink-700 text-white font-medium py-2 px-4 rounded-xl transition-all"
          data-testid="product-cta"
          href="https://www.mercadopago.com.br/chaveiro"
          rel="noopener noreferrer"
          target="_blank"
        >
          Comprar
        </a>
      </div>
    </div>
  </div>
</div>
`;
